### ISSUE TITLE
How to understand “each mini-batch Yi always contains these five training samples of this class” and how to change it.

### ISSUE THREAD

**Issue by swiftest** [Torch-HXM/HybridSN-Pytorch#1]
https://github.com/swiftest/RSSGL/issues/1#issue-1797920668

@Torch-HXM
Thank you for your interest in our research. If you want to use a normal random choose, first change the batch_size setting in the configuration file (i.e. './configs/RSSGL/RSSGL_.....py'), next you need to modify the minibatch_sample function in the base module in the data folder.

The batch_size in the global learning framework should not be set too small, according to your requirements, the modified code is shown below.

def minibatch_sample(gt_mask: np.ndarray, train_indicator: np.ndarray, batch_size):
"""
Args:
gt_mask: 2-D array of shape [height, width]
train_indicator: 2-D array of shape [height, width]
minibatch_size:
Returns:
"""

inds = np.where(train_indicator.ravel() == 1)[0]
np.random.shuffle(inds)
train_inds_list = []
cnt = 0

while True:
    train_inds = np.zeros_like(train_indicator).ravel()
    left = cnt * batch_size  # Suppose batch_size is set to 100, there 529 training samples for IP data set.
    if left >= len(inds):
        break
    right = min((cnt + 1) * batch_size, len(inds))
    fetch_inds = inds[left:right]
    train_inds[fetch_inds] = 1
    train_inds_list.append(train_inds.reshape(train_indicator.shape))
    cnt += 1

return train_inds_list
If you have any further questions, you can email me at [wanghenghrbeu@hrbeu.edu.cn](mailto:wanghenghrbeu@hrbeu.edu.cn).
